src/llv8-inl.h

```diff
@@ -279,6 +279,18 @@ ACCESSOR(JSRegExp, GetSource, js_regexp()->kSourceOffset, String)
 
 ACCESSOR(JSDate, GetValue, js_date()->kValueOffset, Value)
 
+bool JSDate::IsSmi(Error& err) {
```

Comment:
These two helpers seem unnecessary if the call site need to grab the values out either ways?

Reply:
I like the idea of having IsTYPE methods instead of duplicating the tests everywhere. If V8 changes how we check for a type we only need to make changes to one place. But I'd rather have all IsTYPE methods in the Value class, so it resembles the V8 API (https://github.com/nodejs/llnode/pull/247/files#diff-6863f2f9f81092029c00ed6e07b78700R60).

Reply:
But looking again those two don't add much value, since they're just calling `.Check`...

We might want to settle on style here (`TYPE.Check()` vs. `Value.IsTYPE()`), but we don't have to decide in this PR.